src/main/java/tdl/client/runner/CombinedClient.java

```diff
@@ -1,7 +1,5 @@
 package tdl.client.runner;
 
-import tdl.client.ProcessingRules;
-
 
 class CombinedClient {
```

Comment:
I do not think `CombinedClient` is a good abstraction. I think we should merge this with `ChallengeSession`.